### PR TITLE
release: Add script to tag repos

### PR DESCRIPTION
**Pull request by jcvenegas**
Add helper script to tag repos.

- Check all repos are in the same version
- Create annotated tags
- Push tags to the repos

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by jcvenegas** [comment]
Adding tag for test repo as well ( suggested by @chavafg)

File: release/tag_repos.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+tmp_dir=$(mktemp -d -t tag-repos-tmp.XXXXXXXXXX)
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+script_name="$(basename "${BASH_SOURCE[0]}")"
+OWNER=${OWNER:-"kata-containers"}
+PROJECT="Kata Containers"
+PUSH="${PUSH:-"false"}"
+commit="master"
+readonly URL_RAW_FILE="https://raw.githubusercontent.com/${OWNER}"
+#The runtime version is used as deference of latest release
```
Comment (chavafg) [inline release/tag_repos.sh]: s/deference/reference?

**Comment by jcvenegas** [comment]
@chavafg  changes applied.

@bergwolf @laijs @egernst  current 0.0.1 tags created and pushed with this script.

**Comment by chavafg** [comment]
lgtm

File: release/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Release tools #
+
+This directory contains tools for Kata Containers releases.
+
+## tag_repos.sh ##
+
+The `tag_repos.sh` script is used to create tags for the Kata Containers
+repositories. This script make sure that all the repositories are in the
+same version (by checking the VERSION file).
```
Comment (jodh-intel) [inline release/README.md]: You could add backticks for `VERSION` maybe.

File: release/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+It will create an **annotated tag** for the new release version for the
+following repositories:
+
+- agent.
+- proxy.
+- runtime.
+- shim.
+- throttle.
```
Comment (jodh-intel) [inline release/README.md]: "throttler".

File: release/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+Additionally, the script will tag tests and osbuilder repositories to know
```
Comment (jodh-intel) [inline release/README.md]: I'd reword slightly to something like:

> The script will also tag the tests and osbuilder repositories to make it clear which versions of these supporting repositories were used for the release.

File: release/tag_repos.sh [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+commit="master"
+readonly URL_RAW_FILE="https://raw.githubusercontent.com/${OWNER}"
+#The runtime version is used as reference of latest release
+readonly kata_version=$(curl -Ls "${URL_RAW_FILE}/runtime/${commit}/VERSION" | grep -v -P "^#")
+
+function usage() {
+
+	cat <<EOT
+Usage: ${script_name} [options] <args>
+This script creates a new release for ${PROJECT}
```
Comment (jodh-intel) [inline release/tag_repos.sh]: Missing period at the end of the sentence?

**Comment by jcvenegas** [comment]
@jodh-intel  changes applied.

File: release/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Release tools #
+
+This directory contains tools for Kata Containers releases.
+
+## tag_repos.sh ##
+
+The `tag_repos.sh` script is used to create tags for the Kata Containers
+repositories. This script make sure that all the repositories are in the
```
Comment (klynnrif) [inline release/README.md]: Suggested rewrite: This scripts ensures that all the repositories are in the

File: release/README.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+same version (by checking the `VERSION` file).
+
+It will create an **annotated tag** for the new release version for the
```
Comment (klynnrif) [inline release/README.md]: Suggested rewrite: The script creates an **annotated tag** for the new release version for the

File: release/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+following repositories:
+
+- agent.
```
Comment (klynnrif) [inline release/README.md]: Since the items in this list are not complete sentences I suggest removing the period after each item in lines 14-18.

File: release/README.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+- proxy.
+- runtime.
+- shim.
+- throttler.
+
+The script will also tag the tests and osbuilder repositories to make it
```
Comment (klynnrif) [inline release/README.md]: Lines 20-22 suggested rewrite to keep active: The script also tags the tests and osbuilder repositories to make it clear which versions of these supporting repositories are used for the release.

**Comment by jodh-intel** [comment]
Thanks @jcvenegas - once @klynnrif's comments are resolved,

lgtm.

**Comment by jcvenegas** [comment]
@klynnrif  ping , changes applied.

**Comment by jcvenegas** [comment]
@klynnrif   ready to merge, could you ack is good ?
